replication/binlogsyncer.go

```diff
@@ -83,9 +83,13 @@ type BinlogSyncerConfig struct {
 	// read timeout
 	ReadTimeout time.Duration
 
-	// maximum number of attempts to re-establish a broken connection
+	// maximum number of attempts to re-establish a broken connection, zero or negative number means infinite retry.
+	// this configuration will not work if DisableRetrySync is true
 	MaxReconnectAttempts int
 
+	// whether disable re-sync for broken connection
+	DisableRetrySync bool
```

Comment:
setting `MaxReconnectAttempts = 0` is not enough?

Reply:
I prefer changing the way of MaxReconnectAttempts

Reply:
then what about
- MaxReconnectAttempts = 0, no retry
- MaxReconnectAttempts < 0, infinite retry
- MaxReconnectAttempts > 0, retry for MaxReconnectAttempts time

Reply:
ok, for `< 0`, we can change the value to MaxInt

Reply:
MaxReconnectAttempts = MaxInt, no retry
MaxReconnectAttempts = 0, infinite retry
MaxReconnectAttempts > 0, retry for MaxReconnectAttempts time